src/org/traccar/BaseProtocolDecoder.java

```diff
@@ -31,6 +31,25 @@
 
     private final Protocol protocol;
 
+    public long addUnknownDevice(String uniqueId, SocketAddress remoteAddress) {
+        long deviceId = 0;
+        Device newDevice = new Device();
+        newDevice.setName(uniqueId);
+        newDevice.setUniqueId(uniqueId);
+
+        try {
+            Group def = Context.getDataManager().getDefaultGroup();
+            newDevice.setGroupId(def.getId());
+            Context.getDeviceManager().addDevice(newDevice);
+            deviceId = Context.getDeviceManager().getDeviceByUniqueId(uniqueId).getId();
```

Comment:
I think you mean I could get the database ID (if I needed, which I don't) by simply calling `newDevice.getId()` right after `Context.getDeviceManager().addDevice(newDevice)`.  I confirm that.  I was a bit confused here.  (I also forgot to remove this line when I didn't need the database ID anymore.)  Thanks.